beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3SchemeURI.java

```diff
@@ -17,29 +17,27 @@
 
 import static java.lang.String.format;
 
-import java.net.URI;
-import java.net.URISyntaxException;
-
 import com.amazonaws.services.s3.AmazonS3URI;
 
 import com.expediagroup.beekeeper.core.error.BeekeeperException;
 
 public class S3SchemeURI {
 
+  private static final String S3_SCHEME = "s3://";
+  private static final String S3_SCHEME_REGEX = "^s3(a|n):\\/\\/";
+
   private AmazonS3URI amazonS3URI;
 
-  public S3SchemeURI(String housekeepingPath) throws URISyntaxException {
+  public S3SchemeURI(String housekeepingPath) {
     if (!housekeepingPath.startsWith("s3")) {
       throw new BeekeeperException(format("'%s' is not an S3 path.", housekeepingPath));
     }
-    URI uri = URI.create(housekeepingPath);
-    URI s3Uri = new URI("s3", uri.getHost(), uri.getPath(), null);
-    this.amazonS3URI = new AmazonS3URI(s3Uri);
+    String s3Path = housekeepingPath.replaceFirst(S3_SCHEME_REGEX, S3_SCHEME);
```

Comment:
So a scheme replacement is all that is needed? Have you found any docs/messages stating this?

Reply:
Good question, I'm pretty sure they are interchangeable, but a quick look for docs doesn't reveal much.

Reply:
By looking [here|https://cwiki.apache.org/confluence/display/HADOOP2/AmazonS3] and [here|https://www.edureka.co/community/920/difference-between-s3n-s3a-and-s3], my understanding is that `s3a` is going to replace `s3n` and simply abstracts `s3` (which is block-based) to use it as an object-based filesystem. I can't understand why the switch would be necessary though.

Reply:
In a recent AWS support case, I was told that we should only be using `s3://`.

https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-plan-file-systems.html